lib/Runtime/Language/InterpreterStackFrame.cpp

```diff
@@ -8208,6 +8208,12 @@ const byte * InterpreterStackFrame::OP_ProfiledLoopBodyStart(const byte * ip)
         {
             BYTE* buffer = arr->GetBuffer();
             *(T2*)(buffer + index) = (T2)GetRegRaw<T2>(value);
+            return;
+        }
+        if (this->m_functionBody->IsWasmFunction())
```

Comment:
You should do the check and trapping in OP_StArrWasm and OP_LdArrWasm, we don't want to add checks for normal asm.js